src/CustomStateSet.ts

```diff
@@ -14,12 +14,15 @@ export class CustomStateSet extends Set<string> {
   }
 
   add(state: string) {
-    if (!/^--/.exec(state) || typeof state !== 'string') {
+    if (!/^--/.test(state) || typeof state !== 'string') {
       throw new DOMException(`Failed to execute 'add' on 'CustomStateSet': The specified value ${state} must start with '--'.`);
     }
     const result = super.add(state);
     const ref = customStateMap.get(this);
     ref.toggleAttribute(`state${state}`, true);
+    if (ref.part) {
+      ref.part.add(`state--${state}`);
```

Comment:
This should probably have been `state${state}` as current it gives `state----foo`. And it means the tests are broken (I'm not familiar with Mocha, but apparently the `describe()` should be `it()`, also the `div` should be a custom element name, or the `CustomStatesSet` constructor throws ; the tests break when run with `npm run test -- --manual`, I have no idea why they pass using playwright).

Reply:
https://github.com/calebdwilliams/element-internals-polyfill/pull/64